File: src/main/java/kr/co/lunatalk/infra/config/toss/TossPaymentsConfig.java
package kr.co.lunatalk.infra.config.toss;

import lombok.RequiredArgsConstructor;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.client.RestClient;

@Configuration
@RequiredArgsConstructor
public class TossPaymentsConfig {

	private final TossPaymentsProperties tossPaymentsProperties;

	@Bean
	public RestClient tossPaymentsRestClient() {
		return RestClient.builder()
			.baseUrl(tossPaymentsProperties.baseUrl())
			.build();
	}
}

File: src/main/kotlin/kr/co/lunatalk/infra/config/toss/TossPaymentsConfig.kt
package kr.co.lunatalk.infra.config.toss

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.client.RestClient

@Configuration
class TossPaymentsConfig(
    private val tossPaymentsProperties: TossPaymentsProperties
) {

    @Bean
    fun tossPaymentsRestClient(): RestClient =
        RestClient.builder()
            .baseUrl(tossPaymentsProperties.baseUrl)
            .build()
}
